### PR TITLE
PFM-ISSUE-24306 added check wether base is a valid commit or branch

### PR DESCRIPTION
**Pull request by stefancplace**
Resolves [PFM-ISSUE-24306](https://base.cplace.io/pages/tboa5g2lg0enng2lfp1lrdp3v/PFM-ISSUE-24306-Frontend-build-fails-after-initial-branch-off?confirmationMessageIds=rhkbl4xgzql1ifrbi2w6ot6wl#id_wsrqdlqi9cus7fztqfsfa3gnz=newOverview)

`changelog: Frontend-Core: added check wether base is a valid commit or branch [PFM-ISSUE-24306, PR github-actions#55]`

### CODE REVIEW

File: tools/scripts/run-many/run-many.ts [hunk 1]
```diff
@@ -6,10 +6,16 @@ const target = process.argv[2];
 const jobIndex = Number(process.argv[3]);
 const jobCount = Number(process.argv[4]);
 let base = process.argv[5];
+
 // in case base is not a SHA1 commit hash add origin
 if (!/\b[0-9a-f]{5,40}\b/.test(base)) base = 'origin/' + base;
+if(base.includes('0000000000000000')){
+  base='origin/main'
```
Comment (slaven3kopic) [inline tools/scripts/run-many/run-many.ts:13]: it could also be `master`

File: tools/scripts/run-many/run-many.ts [hunk 2]
```diff
@@ -6,10 +6,16 @@ const target = process.argv[2];
 const jobIndex = Number(process.argv[3]);
 const jobCount = Number(process.argv[4]);
 let base = process.argv[5];
+
 // in case base is not a SHA1 commit hash add origin
 if (!/\b[0-9a-f]{5,40}\b/.test(base)) base = 'origin/' + base;
+if(base.includes('0000000000000000')){
```
Comment (slaven3kopic) [inline tools/scripts/run-many/run-many.ts:12]: Hopefully it is always that string 😄

File: tools/scripts/run-many/run-many.ts (hunk 1 above)
Comment (slaven3kopic) [reply tools/scripts/run-many/run-many.ts:13]: Maybe we could add a unit test for this?

Comment (stefancplace) [reply tools/scripts/run-many/run-many.ts]: not sure a unit test makes sense here also I´m not sure wether this solution works at all

**Review by slaven3kopic — APPROVED** [review]

**Pull request merged**